### PR TITLE
Moves speaking verbs to tongues + subtypes, moves wing sprites to wing subtypes, bodypart damage examines to limbs, fixes sign language not working without a tongue

### PR DESCRIPTION
**Pull request by tralezab**
### Moves speaking verbs to tongues + subtypes
Moves species say mod onto tongues, creates any tongues that didn't exist for the say mods they needed to hold.

### moves wing sprites to wing subtypes
Moves the logic of selecting a wing sprite onto subtypes of /functional on the wing type. Now, angel wings bring the holy trait with them, it isn't a special check on flight potions, and we can expand it. (EMPs taking down robowings? Fires burning megamoth wings? Cool stuff)

### bodypart damage examines to limbs
Instead of checking what your species says, it tallies up your limbs and provides the damage description that matches most of your limbs. So for example, If you're mostly human with one augmented part, you take bruises and cuts. If you're mostly robot augmented with one human part, you get robot damage descriptions. Yay!

### fixes sign language working without a tongue
Having no tongue would garble your speech, and this had no interaction with sign language, so you'd be speaking in broken gurgling with perfectly working hands. Now, the sign language component prevents any kind of garbling, since it brings its own garbling for full/missing arms

image

## Why It's Good For The Game

Moving things off of species inherent makes the game expose way more interesting mechanics to play with. It sucks that you can't steal a jellyperson's chirping, since they can get a normal tongue and they'll go back to... chirping! LAME! THAT IS LAME!

Ditto goes for wings, and for limbs, well, having someone be entirely augmented but get descriptions of bleeding because they didn't spawn as an android is kinda lame.

<details>
  <summary>Spoiler warning</summary>
  
  image
  
</details>

## Changelog
:cl:
refactor: Refactored wings, tongues, and some examine messages, hopefully with minimal effect on actual changes. A few more species have tongues, angel wings bring the holy trait with them, and wings have new descriptions. should be the biggest parts of it
/:cl:

### CODE REVIEW

**Comment by tralezab** [comment]
image
admins will enjoy this one

**Comment by itseasytosee** [comment]
AMAZING WORK

File: code/modules/mining/lavaland/tendril_loot.dm [hunk 1]
```diff
@@ -535,25 +535,47 @@
...
+
+/datum/reagent/flightpotion/proc/get_wing_choice(mob/living/carbon/human/needs_wings)
+	var/list/wing_types = needs_wings.dna.species.wing_types.Copy()
+	if(wing_types.len == 1 || !needs_wings.client)
+		return wing_types[1]
+	var/list/radial_wings = list()
+	var/list/name2type = list()
+	for(var/obj/item/organ/external/wings/functional/possible_type as anything in wing_types)
+		var/datum/sprite_accessory/accessory = GLOB.wings_list[possible_type.name] //Gets the datum for every wing this species has, then prompts user with a radial menu
+		var/image/img = image(icon = 'icons/mob/species/wings.dmi', icon_state = "m_wingsopen_[accessory.icon_state]_BEHIND") //Process the HUD elements
```
Comment (Kapu1178) [inline code/modules/mining/lavaland/tendril_loot.dm:566]: please dont hard code this stuff it hurts the souls of refactor/repath makers

**Comment by Time-Green** [comment]
Very cool, I’ll look at it tonight

File: code/__DEFINES/traits.dm [hunk 2]
```diff
@@ -996,3 +997,12 @@ Remember to update _globalvars/traits.dm if you're adding/removing/renaming trai
...
 /// Trait given by echolocation component.
 #define ECHOLOCATION_TRAIT "echolocation"
+
+///without a human having this trait, they speak as if they have no tongue.
+#define TRAIT_SPEAKS_CLEARLY "speaks_clearly"
+
+// specific sources for TRAIT_SPEAKS_CLEARLY
+
+#define SPEAKING_FROM_TONGUE "tongue"
+#define SPEAKING_FROM_HANDS "hands"
```
Comment (Time-Green) [inline code/__DEFINES/traits.dm:1007]: Autodoc these please, like how they act or something

File: code/modules/mining/lavaland/tendril_loot.dm [hunk 3]
```diff
@@ -535,25 +535,47 @@
...
-			to_chat(exposed_carbon, span_userdanger("A terrible pain travels down your back as your wings change shape!"))
+	if(!ishuman(exposed_mob) || exposed_mob.stat == DEAD)
+		return
+	var/mob/living/carbon/human/exposed_human = exposed_mob
+	if(!HAS_TRAIT(exposed_human, TRAIT_CAN_USE_FLIGHT_POTION) || reac_volume < 5 || !exposed_human.dna)
+		if((methods & INGEST) && show_message)
+			to_chat(exposed_human, span_notice("<i>You feel nothing but a terrible aftertaste.</i>"))
+		return
+	var/obj/item/organ/external/wings/current_wings = exposed_human.getorganslot(ORGAN_SLOT_EXTERNAL_WINGS)
+	if(current_wings)
```
Comment (Time-Green) [inline code/modules/mining/lavaland/tendril_loot.dm:546]:
```suggestion
	if(exposed_human.getorganslot(ORGAN_SLOT_EXTERNAL_WINGS))
```

File: code/modules/mining/lavaland/tendril_loot.dm (hunk 1 above)
Comment (Time-Green) [reply code/modules/mining/lavaland/tendril_loot.dm:566]: Yeah please don't hardcode this. The icon gen is quite complicated and can't really be extracted from just the types. I'd rather you spawn them temporarily and generate fetch the icons and put it in like a static list or something so you only need it once.

You can get the icon from external_organ.get_overlay()

File: code/modules/mob/living/carbon/human/human_say.dm [hunk 4]
```diff
@@ -1,5 +1,21 @@
+/mob/living/carbon/human/say(message, bubble_type, list/spans, sanitize, datum/language/language, ignore_spam, forced, filterproof, message_range, datum/saymode/saymode)
```
Comment (Time-Green) [inline code/modules/mob/living/carbon/human/human_say.dm:1]: please please please please doc this and all the parameters 
Yes I know it;s copied but touch is touch

File: code/modules/mob/living/carbon/human/human_say.dm [hunk 5]
```diff
@@ -1,5 +1,21 @@
...
+	if(!HAS_TRAIT(src, TRAIT_SPEAKS_CLEARLY))
+		var/static/regex/tongueless_lower = new("\[gdntke]+", "g")
+		var/static/regex/tongueless_upper = new("\[GDNTKE]+", "g")
+		if(message[1] != "*")
+			message = tongueless_lower.Replace(message, pick("aa","oo","'"))
+			message = tongueless_upper.Replace(message, pick("AA","OO","'"))
+	. = ..()
```
Comment (Time-Green) [inline code/modules/mob/living/carbon/human/human_say.dm:8]:
```suggestion
	return ..()
```

File: code/modules/surgery/organs/external/wings/functional_wings.dm [hunk 6]
```diff
@@ -148,101 +141,36 @@
...
-		if(current && (current.return_pressure() >= ONE_ATMOSPHERE*0.85)) //as long as there's reasonable pressure and no gravity, flight is possible
-			ADD_TRAIT(owner, TRAIT_FREE_FLOAT_MOVEMENT, REF(src))
-			return
-
-	REMOVE_TRAIT(owner, TRAIT_FREE_FLOAT_MOVEMENT, REF(src))
-
-///check if our wings can burn off ;_;
-/obj/item/organ/external/wings/moth/proc/try_burn_wings(mob/living/carbon/human/human)
-	SIGNAL_HANDLER
+/obj/item/organ/external/wings/functional/angel
```
Comment (Time-Green) [inline code/modules/surgery/organs/external/wings/functional_wings.dm:144]: Can you autodoc all these wing types?

File: code/modules/surgery/organs/external/wings/functional_wings.dm [hunk 7]
```diff
@@ -148,101 +141,36 @@
...
-	set_sprite("Burnt Off")
+/obj/item/organ/external/wings/functional/skeleton
+	name = "skeletal wings"
+	desc = "Powered by pure edgy-teenager-notebook-scribblings. Just kidding. But seriously, how do these keep you flying?!"
+	stored_feature_id = "Skeleton"
 
-///heal our wings back up!!
-/obj/item/organ/external/wings/moth/proc/heal_wings(datum/source, heal_flags)
-	SIGNAL_HANDLER
+/obj/item/organ/external/wings/functional/moth //i want this here for more behavior in the future, like moth wings burning off.
```
Comment (Time-Green) [inline code/modules/surgery/organs/external/wings/functional_wings.dm:166]: Ill allow it

**Review by Time-Green — CHANGES_REQUESTED** [review]
I love this refactor! Signing is still real fucking weird, but this doesn't make it worse so whatever

Also seeing you wipe out so many vars from the species datum is beautiful

File: code/modules/surgery/organs/external/wings/functional_wings.dm (hunk 7 above)
Comment (tralezab) [reply code/modules/surgery/organs/external/wings/functional_wings.dm:166]: thankies

File: code/modules/mob/living/carbon/human/human_say.dm (hunk 4 above)
Comment (tralezab) [reply code/modules/mob/living/carbon/human/human_say.dm:1]: https://discord.com/channels/326822144233439242/326831214667235328/1048033813294489722

Comment (Time-Green) [reply code/modules/mob/living/carbon/human/human_say.dm:3]: oh rigth I didnt notice

File: code/modules/mining/lavaland/tendril_loot.dm [hunk 8]
```diff
@@ -535,25 +535,46 @@
...
+
+/datum/reagent/flightpotion/proc/get_wing_choice(mob/living/carbon/human/needs_wings)
+	var/list/wing_types = needs_wings.dna.species.wing_types.Copy()
+	if(wing_types.len == 1 || !needs_wings.client)
+		return wing_types[1]
+	var/list/radial_wings = list()
+	var/list/name2type = list()
+	for(var/obj/item/organ/external/wings/functional/possible_type as anything in wing_types)
+		var/datum/sprite_accessory/accessory = GLOB.wings_list[possible_type.name] //Gets the datum for every wing this species has, then prompts user with a radial menu
+		var/image/img = image(icon = accessory.icon, icon_state = "m_wingsopen_[accessory.icon_state]_BEHIND") //Process the HUD elements
```
Comment (Time-Green) [inline code/modules/mining/lavaland/tendril_loot.dm:565]: Ugggh

**Pull request merged**